firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt

```diff
@@ -26,146 +25,52 @@ import com.google.firebase.ktx.Firebase
 import com.google.firebase.ktx.app
 import com.google.firebase.sessions.api.FirebaseSessionsDependencies
 import com.google.firebase.sessions.api.SessionSubscriber
-import com.google.firebase.sessions.settings.SessionsSettings
+import com.google.firebase.sessions.leader.SessionMaintainerLeader
 import kotlinx.coroutines.CoroutineDispatcher
 
-/** The [FirebaseSessions] API provides methods to register a [SessionSubscriber]. */
-class FirebaseSessions
-internal constructor(
-  private val firebaseApp: FirebaseApp,
+/**
+ * The [FirebaseSessions] API provides methods to register a [SessionSubscriber] and delegates to a
+ * [SessionMaintainer].
+ */
+class FirebaseSessions(
+  firebaseApp: FirebaseApp,
   firebaseInstallations: FirebaseInstallationsApi,
   backgroundDispatcher: CoroutineDispatcher,
   blockingDispatcher: CoroutineDispatcher,
   transportFactoryProvider: Provider<TransportFactory>,
 ) {
-  private val applicationInfo = SessionEvents.getApplicationInfo(firebaseApp)
-  private val sessionSettings =
-    SessionsSettings(
-      firebaseApp.applicationContext,
-      blockingDispatcher,
-      backgroundDispatcher,
-      firebaseInstallations,
-      applicationInfo,
-    )
-  private val timeProvider: TimeProvider = Time()
-  private val sessionGenerator: SessionGenerator
-  private val eventGDTLogger = EventGDTLogger(transportFactoryProvider)
-  private val sessionCoordinator = SessionCoordinator(firebaseInstallations, eventGDTLogger)
-
-  init {
-    sessionGenerator = SessionGenerator(collectEvents = shouldCollectEvents(), timeProvider)
 
-    val sessionInitiateListener =
-      object : SessionInitiateListener {
-        // Avoid making a public function in FirebaseSessions for onInitiateSession.
-        override suspend fun onInitiateSession(sessionDetails: SessionDetails) {
-          initiateSessionStart(sessionDetails)
-        }
-      }
+  private val sessionMaintainer: SessionMaintainer
+  private val tag = "FirebaseSessions"
```

Comment:
Why change the const TAG below and add this? Is this left over from splitting?

Reply:
because it requires a companion object which this class has, however, others do not. I figured since every class is going to need one of these that we'd define it in a uniform way that doesn't require adding companion objects. Thoughts?

Reply:
Ok that's fine. Can you still make them `const` though?

Reply:
Strangely no 
![image](https://github.com/firebase/firebase-android-sdk/assets/426640/fd35b913-f05f-4389-8219-6eb37640e15a)